packages/trpc/server/routers/viewer/bookings/confirm.handler.ts

```diff
@@ -109,30 +113,15 @@ export const confirmHandler = async ({ ctx, input }: ConfirmOptions) => {
     },
   });
 
-  if (booking.userId !== user.id && booking.eventTypeId) {
-    // Only query database when it is explicitly required.
-    const eventType = await prisma.eventType.findFirst({
-      where: {
-        id: booking.eventTypeId,
-        schedulingType: SchedulingType.COLLECTIVE,
-      },
-      select: {
-        users: {
-          select: {
-            id: true,
-          },
-        },
-      },
-    });
+  await checkIfUserIsAuthorizedToConfirmBooking({
+    eventTypeId: booking.eventTypeId,
+    loggedInUserId: user.id,
+    teamId: booking.eventType?.teamId,
+    bookingUserId: booking.userId,
+    userRole: user.role,
+  });
 
-    const membership = booking.eventType?.team?.members.find((membership) => membership.userId === user.id);
-    const isTeamAdminOrOwner =
-      membership?.role === MembershipRole.OWNER || membership?.role === MembershipRole.ADMIN;
-
-    if (eventType && !eventType.users.find((user) => booking.userId === user.id) && !isTeamAdminOrOwner) {
-      throw new TRPCError({ code: "UNAUTHORIZED", message: "UNAUTHORIZED" });
-    }
-  }
+  return;
```

Comment:
Will this return avoid running the rest of the code?

Reply:
sorry forgot to remove this